packages/project-integrity/src/types.ts

```diff
@@ -0,0 +1,75 @@
+/**
+ * Project settings when initializing project integrity.
+ */
+export interface ProjectSettings {
+    /**
+     * Path to the file where integrity data, like hash values, are stored
+     */
+    integrityFilePath: string;
+
+    /**
+     * List of files to create a hash value for
+     */
+    fileList: string[];
+
+    /**
+     * List of content (key->string) to create a hash value for
+     */
+    additionalStringContent?: Content;
+}
+
+/**
+ * Integrity data for a project, stored as JSON in integrity file.
+ */
+export interface Integrity {
+    enabled: boolean;
+    fileIntegrity: FileIntegrity[];
+    contentIntegrity: ContentIntegrity[];
+}
+
+/**
+ * Integrity data for a file.
+ */
+export interface FileIntegrity {
+    filePath: string;
+    hash: string;
```

Comment:
@Klaus-Keller do we plan to keep both hash content and compressed content? Maybe compressed content does job of hash? or maybe adding a settings to include hash can be an option?